plugin/pkg/admission/security/podsecuritypolicy/admission.go

```diff
@@ -120,8 +123,17 @@ func (c *podSecurityPolicyPlugin) Admit(a admission.Attributes) error {
 	}
 
 	pod, ok := a.GetObject().(*api.Pod)
-	// if we can't convert then we don't handle this object so just return
+	// if we can't convert then fail closed since we've already checked that this is supposed to be a pod object.
+	// this shouldn't normally happen during admission but could happen if an integrator passes a versioned
+	// pod object rather than an internal object.
 	if !ok {
+		return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetObject()))
+	}
+
+	// if this is an update, see if we are only updating the ownerRef.  Garbage collection does this
+	// and we should allow it in general, since you had the power to update and the power to delete.
+	// The worst that happens is that you delete something, but you aren't controlling the privileged object itself
```

Comment:
I don't understand this comment. Why does this change require the power to delete? Also what are the implications of changing the owner? (Sorry, not that familiar with GC).

Reply:
ownerref and finalizer fields can require updates during the deletion lifecycle of an object. requiring the actors that update those fields to have PSP permissions in order to remove ownerrefs/finalizers can eternally block deletion of objects.

updates that change only ownerRefs can trigger an object being deleted, but do not affect anything related to the pod's execution or security context

updates that change finalizers can block/unblock an object from being deleted, but do not affect anything related to the pod's execution or security context

for that reason, we don't need to rerun PSP checks on update if only those fields are being touched.